include/boost/numeric/odeint/stepper/adaptive_adams_bashforth_moulton.hpp

```diff
@@ -0,0 +1,196 @@
+		typedef Deriv deriv_type;
+		typedef Time time_type;
+		typedef Resizer resizer_type;
+
+		typedef Algebra algebra_type;
+		typedef Operations operations_type;
+		typedef algebra_stepper_base< Algebra , Operations > algebra_stepper_base_type;
+
+		typedef state_wrapper<state_type> wrapped_state_type;
+		typedef state_wrapper<deriv_type> wrapped_deriv_type;
+
+		typedef detail::adaptive_adams_coefficients<order_value, deriv_type, time_type, algebra_type, operations_type> coeff_type;
+
+		typedef detail::rotating_buffer<state_type, steps> error_storage_type;
+
+		typedef adaptive_adams_bashforth<order_value, state_type, value_type, state_type, value_type, algebra_type, operations_type> aab_type;
+		typedef adaptive_adams_moulton<order_value, state_type, value_type, state_type, value_type, algebra_type, operations_type> aam_type;
+		typedef adaptive_adams_bashforth_moulton< Steps , State , Value , Deriv , Time, Algebra, Operations, Resizer > stepper_type;
+
+		typedef error_stepper_tag stepper_category;
+
+		adaptive_adams_bashforth_moulton(const algebra_type &algebra = algebra_type())
+		:algebra_stepper_base_type( algebra ), m_aab(algebra), m_aam(algebra),
+		m_dxdt_resizer(), m_xerr_resizer(), m_xnew_resizer(),
+		m_coeff()
+		{};
+
+		order_type order()
+		{
+			return order_value + 1;
+		}
+
+		order_type stepper_order()
+		{
+			return order_value + 1;
+		}
+
+		order_type error_order()
+		{
+			return order_value;
+		}
+
+		template<class System>
+		void do_step(System system, state_type & inOut, time_type &t, time_type &dt)
+		{
+			m_xerr_resizer.adjust_size( inOut , detail::bind( &stepper_type::template resize_xerr_impl< state_type > , detail::ref( *this ) , detail::_1 ) );
+
+			do_step(system, inOut, t, dt, m_xerr.m_v);
+		};
+
+		template<class System>
+		void do_step(System system, const state_type & in, time_type &t, state_type & out, time_type &dt)
+		{
+			m_xerr_resizer.adjust_size( in , detail::bind( &stepper_type::template resize_xerr_impl< state_type > , detail::ref( *this ) , detail::_1 ) );
+
+			do_step(system, in, t, out, dt, m_xerr.m_v);
+		};
+
+		template<class System>
+		void do_step(System system, state_type & inOut, time_type &t, time_type &dt, state_type &xerr)
+		{
+			m_xnew_resizer.adjust_size( inOut , detail::bind( &stepper_type::template resize_xnew_impl< state_type > , detail::ref( *this ) , detail::_1 ) );
+
+			do_step(system, inOut, t, m_xnew.m_v, dt, xerr);
+			boost::numeric::odeint::copy( m_xnew.m_v , inOut);
+		};
+
+		template<class System>
+		void do_step(System system, const state_type & in, time_type &t, state_type & out, time_type &dt, state_type &xerr)
+		{
+			do_step_impl(system, m_coeff, in, t, out, dt, xerr);
+
+			system(out, m_dxdt.m_v, t+dt);
+			m_coeff.step(m_dxdt.m_v, t+dt);
+			m_coeff.confirm();
+		};
+
+		template<class System>
+		void do_step_impl(System system, coeff_type coeff, const state_type & in, time_type &t, state_type & out, time_type &dt, state_type &xerr)
```

Comment:
Called from controlled_adams_bashforth_moulton for the same reason as above (coeff should not be confirmed if the step is rejected)